### ISSUE TITLE
Phone or email articles are deletable when set to internal

### ISSUE THREAD

**Issue by YetAnotherGerrit** [zammad/zammad#4788]
### Used Zammad Version

6.0.0.47e1e5715.saas

### Environment

- Installation method: all
- Operating system (if you're unsure: `cat /etc/os-release` ): all
- Database + version:  all
- Elasticsearch version: all
- Browser + version: all

### Actual behaviour

When you set an email to internal, you are able to delete the article.

### Expected behaviour

Only internal notes should be able to be deleted.

### Steps to reproduce the behaviour

Write an e-mail, change the visibility to "internal" a "delete"-button appears.

### I'm sure this is a bug and no feature request or a general question.

yes

**Comment by mantas** [comment]
Please note that this was the desired behavior a long time ago. We may get angry people claiming this is a regression ocnce again :)

https://github.com/zammad/zammad/issues/2853/

**Comment by fliebe92** [comment]
To solve the reported behavior in a proper way, we need to think about another solution. It must be possible to still delete notes, even external ones.

**Comment by dvuckovic** [comment]
This also applies to WhatsApp messages and possibly other channels. Own articles are possible to delete in the configured time frame, but this is misleading as the message can not be deleted on the target channel, similar to how a sent out email cannot be "revoked".

Possible solution: limit this mechanism to notes only.

**Comment by mantas** [comment]
@dvuckovic there's already a mechanism that prevents non-internal communication messages from being deleted.

Maybe Whatsapp is not marked correctly as non-internal communication message?

https://github.com/zammad/zammad/blob/141264e8d3098f48474a26074a736f52d9b93662/app/policies/ticket/article_policy.rb#L31

**Comment by dvuckovic** [comment]
@mantas Like any other channel, it *is* possible to set a WhatsApp message as internal. The delete action shows up then.

**Comment by mantas** [comment]
Maybe we should not allow to set communication that was once external to internal :|

**Comment by dvuckovic** [comment]
@mantas I think this is a separate issue, channels do not seem able to prevent this, you can use it for email articles too.

Let's concentrate here on fixing the availability of the delete article action. In my opinion, it makes no sense to ever be available for anything other than notes.

**Comment by mantas** [comment]
The problem is there're internal note-like articles which technically have non-note type. It should be possible to delete those too.

Long time ago I had added a change that allows to delete internal non-communication articles only. But then I learned that there're systems abusing types taxonomy. They use custom internal & communication types. And expect to be able to delete them!

That's how we arrived to this `communication && !internal` check.

**Comment by YetAnotherGerrit** [comment]
> But then I learned that there're systems abusing types taxonomy. They use custom internal & communication types.

Can you dig up more details on that? And can we offer those people a "clean solution" for their use case?

**Comment by cautopartes** [comment]
I came across this during a training session, when I was explaining that you can only delete **your own notes**, by default ten minutes after creation (and that the time frame can be changed via rails command).

But actually, I can delete also my own _Phone_ and _email_ articles by setting them to _internal_ first.
In my view, this contradicts the proclaimed aim to persevere information and make it as hard as possible to have it removed from agent perspective.

**Comment by dvuckovic** [comment]
@YetAnotherGerrit One use-case I noticed is employed fairly often in the wild (and [that even our documentation mentions](https://docs.zammad.org/en/latest/admin/console/working-on-ticket-articles.html#count-public-notes-toward-slas)) is counting public notes towards SLA. In this case, note article type gets its communication flag set to `true`. Which basically means no more deleting public notes due to the current guard.

We should re-specify this mechanism, as I think the current behavior is too unpredictable and may not align too well with real-life use-cases.

**Comment by YetAnotherGerrit** [comment]
I agree. But I actually want to re-specify more details related to that. I want to redesign the complete "deletion" mechanic. I see synergies with the request to **edit** existing notes as well. I also don't like that it is even possible to delete articles without any trace of what was deleted. Modified (which includes deleted) articles need to be auditable. It's now part of the roadmap.
